src/main/java/com/mineboundteam/minebound/capabilities/PlayerUtilityToggleProvider.java

```diff
@@ -0,0 +1,71 @@
+package com.mineboundteam.minebound.capabilities;
+
+import net.minecraft.core.Direction;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.entity.player.Player;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.CapabilityManager;
+import net.minecraftforge.common.capabilities.CapabilityToken;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.common.util.INBTSerializable;
+import net.minecraftforge.common.util.LazyOptional;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class PlayerUtilityToggleProvider implements ICapabilityProvider, INBTSerializable<CompoundTag> {
+    public static Capability<UtilityToggle> UTILITY_TOGGLE = CapabilityManager.get(new CapabilityToken<>() { });
+
+    private UtilityToggle utilityToggle = null;
+    private final LazyOptional<UtilityToggle> optional = LazyOptional.of(this::createUtilityToggle);
+
+
+    private @NotNull PlayerUtilityToggleProvider.UtilityToggle createUtilityToggle() {
+        if(this.utilityToggle == null) {
+            this.utilityToggle = new UtilityToggle();
+            this.utilityToggle.fire = false;
```

Comment:
You shouldn't need this line to set `fire` to `false` since its default value is already set in `UtilityToggle`. Don't want to set that precedent when it's not needed.